app/product_notices.go

```diff
@@ -77,10 +80,23 @@ func noticeMatchesConditions(config *model.Config, preferences store.PreferenceS
 	}
 
 	// check if current server version is notice range
-	serverVersion, err := semver.NewVersion(model.BuildNumber)
-	if err != nil {
-		mlog.Warn("Skipping server version check, build number is not in semver format", mlog.String("build_number", model.BuildNumber))
-	} else {
+	if cnd.ServerVersion != nil {
+		// clean up BuildNumber to remove release- prefix, -rc suffix and a hash part of the version
+		version := strings.Replace(model.BuildNumber, "release-", "", 1)
+		version = rcStripRegexp.ReplaceAllString(version, `$1$3`)
+		versionParts := strings.Split(version, ".")
+		var versionPartsOut []string
+		for _, part := range versionParts {
+			if _, err := strconv.ParseInt(part, 10, 16); err == nil {
+				versionPartsOut = append(versionPartsOut, part)
+			}
+		}
+		version = strings.Join(versionPartsOut, ".")
```

Comment:
This would probably look better as a standalone function.

Reply:
@streamer45 - agreed. will do.